flax/linen/module.py

```diff
@@ -1320,6 +1321,42 @@ def __call__(self, x):
     scope = core.bind(variables, rngs=rngs, mutable=mutable)
     return self.clone(parent=scope)
 
+  def unbind(self: M) -> Tuple[M, VariableDict]:
+    """Returns an unbound copy of this Module and its variables.
+
+    ``unbind`` is useful for creating a stateless version of a bound Module.
+    A common use case araises when you want extract a submodule defined inside
+    ``setup()``, since ``setup()`` is only called when the Module is bound, a
+    simple way to get hold of the submodule and its corresponding variables is
+    to ``bind`` the parent Module temporarily and then ``unbind`` the desired
+    submodule::
```

Comment:
Nit:

- sub-Module (we should do this across all docs)
- arises (typo)
- There's a repetition of "You want [to] extract a submodule... to get hold of the submodule...]
- Maybe replace "A common use arises with "An example of a common use case..."
  - Otherwise you have a very complex sentence (in an `unbind` explanation) that may be hard to follow.

Example:

```suggestion
  def unbind(self: M) -> Tuple[M, VariableDict]:
    """Returns an unbound copy of a Module and its variables.

    ``unbind`` helps create a stateless version of a bound Module.

    An example of a common use case: to extract a sub-Module defined inside 
    ``setup()`` and its corresponding variables: 1) temporarily ``bind`` the parent
    Module; and then 2) ``unbind`` the desired sub-Module. (Recall that ``setup()``
    is only called when the Module is bound.)::
```

cc @marcvanzee